bundles/core/org.eclipse.smarthome.core.library/src/main/java/org/eclipse/smarthome/core/library/types/GenericComplexType.java

```diff
@@ -0,0 +1,90 @@
+/**
+ * Copyright (c) 2014-2015 openHAB UG (haftungsbeschraenkt) and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+
+package org.eclipse.smarthome.core.library.types;
+
+import java.util.Formatter;
+import java.util.SortedMap;
+import java.util.TreeMap;
+
+import org.eclipse.smarthome.core.library.types.StringType;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.ComplexType;
+import org.eclipse.smarthome.core.types.PrimitiveType;
+import org.eclipse.smarthome.core.types.State;
+
+
+/**
+ * This type can be used for items that are dealing with telephony functionality.
+ * 
+ * @author Thomas.Eichstaedt-Engelen / Initially CallType
+ * @author Gaël L'hopital - port to Eclipse SmartHome
+ * 
+ */
+public class GenericComplexType implements ComplexType, Command, State {
+	
+	private SortedMap<String, PrimitiveType> typeDetails = new TreeMap<String, PrimitiveType>();
+	
+	public GenericComplexType(StringType...elements) {
+		for (StringType element : elements) {
+			String[] parts = element.toString().split("=");
+			if (elements.length == 2) {
```

Comment:
I'm ok with the Map constructor as well.
But it does not solve the issue that we need to define a string serialization format. The methods ```toString``` and ```valueOf(String)``` must do the (de-)serialization. This is needed for all communication - remember that states are e.g. passed as a String argument through the REST API. In the CallType we had "##" as a separator. This is probably not applicable for all complex types. Is there a way to define a suitable separator by an item type...?